Fixed ``CancelledError`` from one cleanup context stopping other contexts from completng -- by :user:`Dreamsorcerer`.
